### PR TITLE
resources and data sources for CSI volumes and plugins

### LINKED ISSUE #102: Enhancement:  Support for Nomad Volumes
With support for CSI storage, Terraform provider for Nomad needs to support registration of Volumes.  Full parameter set for Volumes are located in the [Nomad Volume Register documentation](https://www.nomadproject.io/docs/commands/volume/register/).

Terraform v0.12.24
provider.nomad v1.4.5
Nomad v0.11.0

### PR DESCRIPTION
**Pull request by cgbaker**
new resource for registering Nomad CSI Volumes
in addition, there are data sources for querying CSI plugins and volumes
some of these are very useful when orchestrating the installation of CSI plugins and controllers, as this is a dependency that must be satisfied before volumes can be registered

closes #102

### CODE REVIEW

File: nomad/data_source_volumes.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	}
+}
+
+func volumesDataSourceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(ProviderConfig).client
+
+	ns := d.Get("namespace").(string)
+	if ns == "" {
+		ns = "default"
+	}
```
[the remark this answers is not in the document]
Comment (cgbaker) [reply nomad/data_source_volumes.go:63]: good point. i copied that from one of the other ones.

File: nomad/provider.go [hunk 2]
```diff
@@ -70,7 +70,10 @@ func Provider() terraform.ResourceProvider {
 			"nomad_deployments": dataSourceDeployments(),
 			"nomad_job":         dataSourceJob(),
 			"nomad_namespaces":  dataSourceNamespaces(),
+			"nomad_plugin":      dataSourcePlugin(),
+			"nomad_plugins":     dataSourcePlugins(),
```
[the remark this answers is not in the document]
Comment (cgbaker) [reply nomad/provider.go:74]: no, they're for "dynamic" plugins, but currently the only type of dynamic plugin is a CSI plugin. i'm not sure what the future will be there, but i wanted to honor the API as much as possible.

[the remark this answers is not in the document]
Comment (lgfa29) [reply nomad/provider.go:74]: Ah, makes sense. I thought the API was CSI specific 👍

File: website/nomad.erb [hunk 3]
```diff
@@ -55,6 +64,9 @@
             <li<%= sidebar_current("docs-nomad-resource-sentinel-policy") %>>
               <a href="/docs/providers/nomad/r/sentinel_policy.html">nomad_sentinel_policy</a>
             </li>
+            <li<%= sidebar_current("docs-nomad-resource-volume-policy") %>>
+              <a href="/docs/providers/nomad/r/sentinel_volume.html">nomad_volume</a>
```
Comment (cgbaker) [inline website/nomad.erb:68]:
```suggestion
              <a href="/docs/providers/nomad/r/volume.html">nomad_volume</a>
```

**Pull request merged**